core-libs/setup/ssr/optimized-engine/optimized-ssr-engine.ts

```diff
@@ -80,14 +90,26 @@ export class OptimizedSsrEngine {
       if (typeof value === 'function') {
         return value.toString();
       }
+      if (this.isServerLogger(value)) {
+        return value.constructor.name;
+      }
```

Comment:
[MINOR]

What about simply:
```suggestion
      if (value?.constructor?.name) {
        return value.constructor.name;
      }
```

If we really have to use `isServerLogger` (or `isExpressServerLogger`), I think it shouldn't be a method of `OptimizedSsrEngine`. It's not its responsibility. I'd rather move this type guard to `express-server-logger.ts`

Anyway IMHO `if (value?.constructor?.name)` should suffice. What do you think?

Reply:
Since `replacer` in `JSON.stringify()` works recursively, if we put there such a condition and put an object as a value to be stringified, we will end with "Object" as an output from this method. (which is our case):
![image](https://github.com/SAP/spartacus/assets/28891782/a159ff42-4f8a-4fa4-a825-80e46c756429)

I'll move type guard to `express-server-logger.ts`

Reply:
`logOpotions` method has been refactored and the type guard has been removed as it's no longer needed.

Reply:
done in https://github.com/SAP/spartacus/pull/17538